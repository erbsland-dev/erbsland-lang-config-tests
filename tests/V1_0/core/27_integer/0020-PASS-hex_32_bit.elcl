# Erbsland Configuration Language Test File
[main]
integer 01: 0x0
integer 02: 0x1
integer 03: -0x1
integer 04: +0x1
integer 05: 0xa
integer 06: -0xa
integer 07: +0xa
integer 08: 0x0000009f    # Zero prefix is allowed
integer 09: -0x0000009f   # Zero prefix is allowed
integer 10: 0x7fffffff  # 32 bit maximum.
integer 11: -0x80000000 # 32 bit minimum.
integer 12: 0x1'a         # Makes no sense, but is valid
integer 13: -0x1'a        # Makes no sense, but is valid
integer 14: 0x1f'e4       # Makes no sense, but is valid
integer 15: -0x1f'e4      # Makes no sense, but is valid
integer 16: 0x1a3'c       # Makes no sense, but is valid
integer 17: -0x1a3'c      # Makes no sense, but is valid
integer 18: 0x1'2'd       # Makes no sense, but is valid
integer 19: -0x1'2'd      # Makes no sense, but is valid
integer 20: 0xab'cd'02
integer 21: -0xab'cd'02
integer 22: 0x7fff'ffff
integer 23: -0x8000'0000
integer 24: 0x7'f'f'f'f'f'f'f  # Buffer test
integer 25: -0x8'0'0'0'0'0'0'0  # Buffer test
integer 26: -0x0          # Makes no sense, but is valid
integer 27: 0x0abcdef
integer 28: 0X0ABCDEF
integer 29: 0X0aBcDeF
integer 30: 0x0AbCdEf
