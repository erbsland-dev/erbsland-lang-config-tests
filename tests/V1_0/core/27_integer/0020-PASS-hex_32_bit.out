main = SectionWithNames()
main.integer_01 = Integer(0)
main.integer_02 = Integer(1)
main.integer_03 = Integer(-1)
main.integer_04 = Integer(1)
main.integer_05 = Integer(10)
main.integer_06 = Integer(-10)
main.integer_07 = Integer(10)
main.integer_08 = Integer(159)
main.integer_09 = Integer(-159)
main.integer_10 = Integer(2147483647)
main.integer_11 = Integer(-2147483648)
main.integer_12 = Integer(26)
main.integer_13 = Integer(-26)
main.integer_14 = Integer(8164)
main.integer_15 = Integer(-8164)
main.integer_16 = Integer(6716)
main.integer_17 = Integer(-6716)
main.integer_18 = Integer(301)
main.integer_19 = Integer(-301)
main.integer_20 = Integer(11259138)
main.integer_21 = Integer(-11259138)
main.integer_22 = Integer(2147483647)
main.integer_23 = Integer(-2147483648)
main.integer_24 = Integer(2147483647)
main.integer_25 = Integer(-2147483648)
main.integer_26 = Integer(0)
main.integer_27 = Integer(11259375)
main.integer_28 = Integer(11259375)
main.integer_29 = Integer(11259375)
main.integer_30 = Integer(11259375)
